tlslite/extensions.py

```diff
@@ -529,6 +529,71 @@ def parse(self, p):
 
         return self
 
+
+class SupportedVersionsExtension(TLSExtension):
+    """
+    This class handles the SupportedVersion extensions used in TLS 1.3.
+
+    See draft-ietf-tls-tls13.
+
+    :vartype extType: int
+    :ivar extType: numeric type of the Supported Versions extension, i.e. 43
+
+    :vartype extData: bytearray
+    :ivar extData: raw representation of the extension data
+
+    :vartype versions: list of tuples
+    :ivar versions: list of supported protocol versions; each tuple has two
+        one byte long integers
+    """
+
+    def __init__(self):
+        """Create an instance of SupportedVersionsExtension."""
+        super(SupportedVersionsExtension, self).__init__(extType=
+                                                         ExtensionType.
+                                                         supported_versions)
+        self.versions = None
+
+    @property
+    def extData(self):
+        """
+        Return raw encoding of the extension
+
+        :rtype: bytearray
+        """
+        if self.versions is None:
+            return bytearray(0)
+
+        writer = Writer()
+        # elelements 1 byte each, overall length encoded in 1 byte
+        writer.addVarTupleSeq(self.versions, 1, 1)
```

Comment:
yes, they are :)

the length of the tuple is implicit, not passed in as parameter, the parameters specify in how many bytes each element of the tuple has to be encoded in and how long is the length field.

So the call means, encode `self.versions` elements in list that will have its length saved in one byte (2nd "1") and every element in all the tuples should be encoded in one byte (1st "1")